paper-input-container.html

```diff
@@ -553,17 +559,18 @@
     _onValueChanged: function(event) {
       var input = event.target;
 
-      // Problem: if the input is required but has no text entered, we should
-      // only validate it on blur (so that an empty form doesn't come up red
-      // immediately; however, in this handler, we don't know whether this is
-      // the booting up value or a value in the future. I am assuming that the
-      // case  we care about manifests itself when the value is undefined.
-      // If this causes future problems, we need to do something like track whether
-      // the iron-input-ready event has fired, and this handler came before that.
-
-      if (input.value === undefined) {
+      // Paper-input treats a value of undefined differently at startup than
+      // the rest of the time (specifically: it does not validate it at startup, but
+      // it does after that. If this is in fact the bootup case, ignore validation,
+      // just this once.
+      if (this.__isFirstValueUpdate && input.value === undefined) {
```

Comment:
nit: WDYT of wrapping the previous if condition?
```js
if (this.__isFirstValueUpdate) {
  this.__isFirstValueUpdate = false;
  if (input.value === undefined) {
    return;
  }
}

Reply:
Very pro this change, thanks